src/Microsoft.AspNetCore.Server.Testing/Deployers/NanoDeployer/NanoDeployer.cs

```diff
@@ -0,0 +1,251 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using System.Diagnostics;
+using System.IO;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Logging;
+using System.Text;
+
+namespace Microsoft.AspNetCore.Server.Testing
+{
+    public class NanoDeployer : ApplicationDeployer
+    {
+        /// <summary>
+        /// Example: If the share path is '\\foo\bar', then this returns the full path to the
+        /// deployed folder. Example: '\\foo\bar\048f6c99-de3e-488a-8020-f9eb277818d9'
+        /// </summary>
+        private string _deployedFolderPathInFileShare;
+        private readonly NanoDeploymentParameters _deploymentParameters;
+        private string _remotePSSessionHelperScript;
+
+        public NanoDeployer(DeploymentParameters deploymentParameters, ILogger logger)
+            : base(deploymentParameters, logger)
+        {
+            _deploymentParameters = DeploymentParameters as NanoDeploymentParameters;
+            if (_deploymentParameters == null)
```

Comment:
Agreed, any reason not to have NanoDeploymentParameters in the constructor itself?